src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -62,16 +63,35 @@ public static async Task RetrieveClaimsAsync(LdapSettings settings, ClaimsIdenti
             {
                 // Example distinguished name: CN=TestGroup,DC=KERB,DC=local
                 var groupDN = $"{Encoding.UTF8.GetString((byte[])group)}";
-                var groupCN = groupDN.Split(',')[0].Substring("CN=".Length);
+
+                if (!TryGetGroupCN(groupDN, out var groupCN))
```

Comment:
There aren't any unit tests for this area. So this change could be reverted if we aren't confident.

Reply:
cc @Tratcher

Reply:
This will conflict with https://github.com/dotnet/aspnetcore/pull/45649/files. Revert changes in this file.